e2etest/scenario_helpers.go

```diff
@@ -541,11 +541,82 @@ func (scenarioHelper) generateAzureFilesFromList(c asserter, shareURL azfile.Sha
 }
 
 func (s scenarioHelper) enumerateShareFileProperties(a asserter, shareURL azfile.ShareURL) map[string]*objectProperties {
+	var dirQ []azfile.DirectoryURL
+	result := make(map[string]*objectProperties)
+
+	root := shareURL.NewRootDirectoryURL()
+	dirQ = append(dirQ, root)
+	for i := 0; i < len(dirQ); i++ {
+		currentDirURL := dirQ[i]
+		for marker := (azfile.Marker{}); marker.NotDone(); {
+			lResp, err := currentDirURL.ListFilesAndDirectoriesSegment(context.TODO(), marker, azfile.ListFilesAndDirectoriesOptions{})
+			a.AssertNoErr(err)
+
+			// Process the files and folders we listed
+			for _, fileInfo := range lResp.FileItems {
+				fileURL := currentDirURL.NewFileURL(fileInfo.Name)
+				fgpr, err := fileURL.GetProperties(context.TODO())
+				a.AssertNoErr(err)
+
+				// Construct the properties object
+				fileSize := fgpr.ContentLength()
+				creationTime, err := time.Parse(azfile.ISO8601, fgpr.FileCreationTime())
+				a.AssertNoErr(err)
+				lastWriteTime, err := time.Parse(azfile.ISO8601, fgpr.FileLastWriteTime())
+				a.AssertNoErr(err)
+				contentHeader := fgpr.NewHTTPHeaders()
+				h := contentHeaders{
+					cacheControl:       &contentHeader.CacheControl,
+					contentDisposition: &contentHeader.ContentDisposition,
+					contentEncoding:    &contentHeader.ContentEncoding,
+					contentLanguage:    &contentHeader.ContentLanguage,
+					contentType:        &contentHeader.ContentType,
+					contentMD5:         contentHeader.ContentMD5,
+				}
+				fileAttrs := fgpr.FileAttributes()
+				filePermissions := fgpr.FilePermissionKey()
```

Comment:
In answer to your offline question: the key is not the thing that we need, in the test harness. What we actually need, is the SDDL string that the key is the key for.  Basically the way AzCopy works with permissions is like this:
* When uploading from Windows to Azure Files, ask Windows for the SDDL string that describes the security info or the file or directory.
* Do some search and replace operations on that for certain short names and replace them with real SIDs that belong to the source location. 
* Send the resulting SDDL to Azure Files.  
* Azure Files Service does a thing where it takes the SDDL and assigns a key to it, and then when you read, its the key that you get back. (That's what you are doing here).  Within a given File Share, all uploads of the same SDDL string will map to the same key at download time.  So typically, when downloading as you are doing here, you'll get many items with the same key. So in theory, you can do some caching to look up the SDDL for each key just once.
* (In the main AzCopy code) if we want to set the permissions on a downloaded file, that we are saving to disk, we just pass Windows the SDDL that we got by looking up the key.

You can see examples of all this if you search for SDDL in the main AzCopy code.